java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowReader.java

```diff
@@ -242,6 +247,9 @@ protected void loadDictionary(ArrowDictionaryBatch dictionaryBatch) {
         VectorBatchAppender.batchAppend(vector, deltaVector);
       }
       return;
+    } else if (validateReplacements && getClass() == ArrowFileReader.class) {
+      throw new IllegalStateException("Replacement dictionaries are not supported in " +
+          "the IPC file format. Dictionary ID: " + dictionary.getEncoding().getId());
```

Comment:
Would it be possible to include these new changes/updates in the IPC documentation file -> docs/source/java/ipc.rst#reading-writing-ipc-formats?

Reply:
> Would it be possible to include these new changes/updates in the IPC documentation file -> docs/source/java/ipc.rst#reading-writing-ipc-formats?

This is not an update or change, it works according to specs, but it will be helpful to present information in a cleaner manner.

Reply:
This is better as a separate issue. But yes, others have noted that the language about what is and is not allowed is very confusing.